### PR TITLE
SA api consumption Giphy responses with gifs

### PR DESCRIPTION
**Pull request by Sergio-Azcona**
In this repo: 
- created (in _./spec/fixture/giphy_) json files which store responses from Giphy (got responses using **Postman**) 
- used **WebMock** to test for Giphy response for instances where gif data exists (i.e. _Happy Path_ response)
  - more specifically: testing for top level and second level keys, and their values' datatype (ex. string, integer, Array, Dictionary,  etc)

Note: after creating the fixture and first test, I realized that my local ruby and rails versions were more recent than the version the team agreed on. Therefore, I downgraded the local rails & ruby versions to be in compliance; now running:
- ruby "2.7.4"
- gem 'rails', '~> 5.2.8', '>= 5.2.8.1'

While downgrading made changes to the **Gem** file and recreated the **Gemfile.lock** (thus, you may see a conflict with _main_ in relation to these files)  

**ALL tests GREEN at push**

### CODE REVIEW

File: Gemfile
Comment (jlweave) [inline Gemfile:1]: cleaned up gem file

File: spec/fixtures/services/giphy/happy_path/full_phrase_response.json
Comment (jlweave) [inline spec/fixtures/services/giphy/happy_path/full_phrase_response.json:1]: is this json something that we can put in the gitignore?

**Review by jlweave — APPROVED** [review]

File: spec/fixtures/services/giphy/happy_path/full_phrase_response.json
Comment (Sergio-Azcona) [reply spec/fixtures/services/giphy/happy_path/full_phrase_response.json:1]: Hi, I did a little research on this and did not find anything that answered the Q head on. I then plugged the Q into our buddy _Chat_ and it noted that json file would not be commented out bc that data is used to run the tests... so anyone else pulling it down would need to have the json files as well. Makes sense to me.

The file is long but that is the standard Giphy response when no limit is set. I will likely limit the return, but for now I do want to test for instances when no limit is set.

What do you think, @jlweave ?
What does everyone else think? @Eldridge-Turambi, @bflanagan138 , @devin-p-lay

Comment (jlweave) [reply spec/fixtures/services/giphy/happy_path/full_phrase_response.json:1]: Okay sounds good and makes sense! Thanks for looking into it!

**Review by bflanagan138 — APPROVED** [review]
We've discussed most of these changes already and modified our group expected behaviours / dynamics to suit. All good in this PR, approved

**Pull request merged**